src/libraries/Common/src/System/Net/SocketAddress.cs

```diff
@@ -122,21 +133,33 @@ internal SocketAddress(IPAddress ipAddress)
 #pragma warning restore CS0618
 
                 Debug.Assert(ipAddress.AddressFamily == AddressFamily.InterNetwork);
-                SocketAddressPal.SetIPv4Address(Buffer, address);
+                SocketAddressPal.SetIPv4Address(InternalBuffer, address);
             }
         }
 
         internal SocketAddress(IPAddress ipaddress, int port)
             : this(ipaddress)
         {
-            SocketAddressPal.SetPort(Buffer, unchecked((ushort)port));
+            SocketAddressPal.SetPort(InternalBuffer, unchecked((ushort)port));
         }
 
         internal SocketAddress(AddressFamily addressFamily, ReadOnlySpan<byte> buffer)
         {
-            Buffer = buffer.ToArray();
-            InternalSize = Buffer.Length;
-            SocketAddressPal.SetAddressFamily(Buffer, addressFamily);
+            InternalBuffer = buffer.ToArray();
+            InternalSize = InternalBuffer.Length;
+            SocketAddressPal.SetAddressFamily(InternalBuffer, addressFamily);
+        }
+
+        /// <summary>This represents underlying memory that can be passed to native OS calls.</summary>
+        /// <remarks>
+        /// This memory can be invalidated if <see cref="Size"/> is changed or if the SocketAddress is used in another receive call.
```

Comment:
With the design decisions we made, the actual InternalBuffer now never changes, right?

Reply:
yes.